src/common/helpers/config.js

```diff
@@ -16,19 +16,47 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-import {readFileSync} from 'fs';
+import {existsSync, readFileSync} from 'fs';
+import {normalize} from 'path';
 
 
-/* Pull in environment-specific configuration. */
-const env = process.env.NODE_ENV || 'development'; // eslint-disable-line no-process-env,max-len
+let configContents;
 
-let configFileBasename = 'config';
-if (env === 'test') {
-	configFileBasename = 'test';
+// Use Default configuration (for Docker)
+function useDefaultConfig() {
+	/* Pull in environment-specific configuration. */
+	const env = process.env.NODE_ENV || 'development'; // eslint-disable-line no-process-env,max-len
+
+	let configFileBasename = 'config';
+	if (env === 'test') {
+		configFileBasename = 'test';
+	}
+	configContents =
+			readFileSync(`config/${configFileBasename}.json`);
 }
 
-const configContents =
-	readFileSync(`config/${configFileBasename}.json`);
+function checkConfigOverwrite() {
+	const args = process.argv;
+	const configIndex = args.indexOf('--config');
+
+	// Check for '--config' followed by 'configPathFile'
+	if (configIndex !== -1 && configIndex < args.length - 1) {
+		let configFilePath = args[configIndex + 1];
+		configFilePath = normalize(configFilePath);
+		if (existsSync(configFilePath)) {
+			configContents = readFileSync(configFilePath);
```

Comment:
The `readFileSync` method can throw an error.
Therefore it should be in a try…catch block, either:
- just for this line
- for the whole `checkConfigOverwrite` method or
- line 58 where you call the method.

Reply:
Yeah true! As I'm checking if the path exists using `existsSync(configFilePath)`, I thought it wasn't needed, but it's actually required since `existsSync` gives true for folders too, so `readFileSync` fails.

Reply:
Correct!
There could also be an unexpected error while reading the file (due to an operating system error or some other surprise for example), and we don't want to possibly prevent the application from launching.